Category button styles

Daste745 opened the pull request:
<!--
	Thank you for contributing to Discord Tickets.
	If you haven't already, please read the CONTRIBUTING guidelines (https://github.com/discord-tickets/.github/blob/main/CONTRIBUTING.md) before creating a pull request.
	Unless this pull request is for something minor like a fixing a typo, you should create an issue first.
-->

**Versioning information**

<!-- Please select **one** by replacing the space with an `x`: `[X]` -->

- [ ] This includes major changes (breaking changes)
- [x] This includes minor changes (minimal usage changes, minor new features)
- [ ] This includes patches (bug fixes)
- [ ] This does not change functionality at all (code refactoring, comments)

**Changes made**

- Added a `buttonStyle` field to the `Category` model
- Set the button's style from the `buttonStyle` field value instead of always using `Secondary`
- Use `Secondary` (2) button style by default

**TODO**
- [x] Migrations for postgres
- [x] Migrations for mysql
- [ ] Documentation

**Confirmations**

<!-- Select **all that apply** by replacing the space with an `x`: `[X]` -->

- [ ] I have updated related documentation (if necessary)
- [ ] My changes use consistent code style
- [ ] My changes have been tested and confirmed to work

Daste745 committed:
feat(panels): Add configurable button styles for categories

Daste745 committed:
feat(categories): Allow setting buttonStyle via PATCH endpoint

Daste745 committed:
chore: Install @discord-tickets/settings from git

Daste745 committed:
chore: Add a compose file for creating DB migrations

Daste745 committed:
chore: Add category button style migrations for mysql and postgresql

Daste745 committed:
chore: Bump @discord-tickets/settings

Daste745 commented:
@HKGx raczej ready. Bundlujemy frontend z https://github.com/strata-czasu/discord-tickets-portal/pull/1, więc nawet nie musimy nic tam mergować. Zrobiłem migracje do mysql+postgres i przetestowałem czy działa - tak, działa.

Po mergu *powinien* zbudować się docker image i wejść na ghcr, ale podejrzewam że coś po drodze się wywali.

HKGx commented on package.json:
#feat/category-button-styles is probably wrong

HKGx reviewed (changes requested):
looks good, one change needed

Daste745 committed:
chore: Use @discord-tickets/settings from the main branch

HKGx reviewed (approved):
lgtm imo @Daste745 push if you're ready

Daste745 committed:
chore: Install dev dependencies during Dockerfile build

This lets us build the @discord-tickets/settings package directly from
git without having to publish a prebuilt version to a registry.
